worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java

```diff
@@ -159,22 +161,31 @@ private void loadRequests(JobData jobData, Job job, Map<Long, CoverageSummary> p
             // Queue a patient
             CoverageSummary patient = patientEntries.next().getValue();
             assert job.getContract() != null;
+
             contractData.addEobRequestHandle(processPatient(contractData.getFhirVersion(),
                     patient, job.getContract(), jobData));
 
             // Periodically check if cancelled
             if (++numQueued % cancellationCheckFrequency == 0) {
+
+                // Update progress periodically as well
+                this.jobChannelService.sendUpdate(jobUuid, JobMeasure.PATIENT_REQUEST_QUEUED, numQueued);
+                jobProgressService.updateProgress(jobUuid);
```

Comment:
Don't do writes directly through jobProgress.  Writes should only be done by sending messages.  I should probably split the Interface then, with one interface for reads more widely used and the write interface used only for the implementation of the Channel Service.